packages/eslint-plugin/lib/rules/no-nested-await.js

```diff
@@ -90,17 +94,15 @@ module.exports = {
         if (!isAwaitAllowedInNode(parent)) {
```

Comment:
The part above this looks unnecessarily opaque to me. Can it be simplified and/or explained (e.g., "_we allow `await …;` expression statements and `identifierName = await …;` assignments, but not update-assignment like `identifierName += await …;` or nesting in other structures such as array/object literals or argument lists_")?

Reply:
I've added an explanatory comment explaining this function's allowlist.